### PR TITLE
Enable private field promotion

### PR DESCRIPTION
**Pull request by goderbauer**
New feature in upcoming Dart 3.2. See https://github.com/dart-lang/language/issues/2020. Feature is enabled by bumping the min SDK version to 3.2.

Part of https://github.com/flutter/flutter/issues/134476.

### CODE REVIEW

**Comment by goderbauer** [comment]
Blocked on https://github.com/dart-lang/sdk/issues/53505.

**Review by chinmaygarde — APPROVED** [review]
LGTM after the roll and presubs.

**Comment by goderbauer** [comment]
Now waiting for https://github.com/dart-lang/sdk/commit/b555e71ecdcca112f449da1656b273663aa65d83 to roll in.

**Pull request merged**
